sdk/examples/how_tos/native_tokens/destroy_foundry.rs

```diff
@@ -34,12 +38,50 @@ async fn main() -> Result<()> {
         let foundry_count = balance.foundries().len();
         println!("Foundries before destroying: {foundry_count}");
 
+        let token_id = TokenId::from(*foundry_id);
+
         // Set the stronghold password
         wallet
             .set_stronghold_password(std::env::var("STRONGHOLD_PASSWORD").unwrap())
             .await?;
 
-        let transaction = account.burn(*foundry_id, None).await?;
+        // Check if all tokens are melted.
+        let native_tokens: Option<&NativeTokensBalance> = balance
+            .native_tokens()
+            .iter()
+            .find(|native_token| *native_token.token_id() == token_id);
+        if let Some(native_token) = native_tokens {
+            let output = account.get_foundry_output(token_id).await?;
+            if native_token.available() != output.as_foundry().token_scheme().as_simple().circulating_supply() {
```

Comment:
Oof, is there really no cleaner way than this?

Reply:
I dont think so no 😢